crates/bevy_picking/src/hover.rs

```diff
@@ -275,3 +275,266 @@ fn merge_interaction_states(
         new_interaction_state.insert(*hovered_entity, new_interaction);
     }
 }
+
+/// A component that allows users to use regular Bevy change detection to determine when the pointer
+/// enters or leaves an entity. Users should insert this component on an entity to indicate interest
+/// in knowing about hover state changes.
+///
+/// This is similar to the old Bevy [`Interaction`] component, except that it only tracks
+/// hover state, not button presses or other interactions.
+///
+/// The component's boolean value will be `true` whenever the pointer is currently hovering over the
+/// entity, or any of the entity's children. This is consistent with the behavior of the CSS
+/// `:hover` pseudo-class, which applies to the element and all of its children.
+///
+/// The contained boolean value is guaranteed to only be mutated when the pointer enters or leaves
+/// the entity, allowing Bevy change detection to be used efficiently. This is in contrast to the
+/// [`HoverMap`] resource, which is updated every frame.
+///
+/// Typically, a simple hoverable entity or widget will have this component added to it. More
+/// complex widgets can have this component added to each hoverable part.
+///
+/// The computational cost of keeping the `IsHovered` components up to date is relatively cheap,
+/// and linear in the number of entities that have the `IsHovered` component inserted.
+///
+/// [`Interaction`]: https://docs.rs/bevy/0.15.0/bevy/prelude/enum.Interaction.html
+#[derive(Component, Copy, Clone, Default, Eq, PartialEq, Debug, Reflect)]
+#[reflect(Component, Default, PartialEq, Debug, Clone)]
+pub struct IsHovered(pub bool);
```

Comment:
Possibly this naming could be confusing because `Is-` components in bevy tend to be unit structs used as markers? It seems natural to use a `With<IsHovered>` query expecting it to return only hovered UI nodes.